owncloudApp/src/main/java/com/owncloud/android/authentication/AccountAuthenticator.java

```diff
@@ -198,6 +201,31 @@ public Bundle getAuthToken(AccountAuthenticatorResponse accountAuthenticatorResp
         return prepareBundleToAccessLoginActivity(accountAuthenticatorResponse, account, authTokenType, options);
     }
 
+    /**
+     * Check if the client has expired or not.
+     * If the client has expired, we can not refresh the token and user needs to re-authenticate.
+     *
+     * @return true if the client is still valid
+     */
+    private boolean clientSecretIsValid(AccountManager accountManager, Account account) {
```

Comment:
Shouldn't the server take care of this?

Reply:
If we take care of it, we can save at least 2 network calls that will fail otherwise.

Reply:
Ok and what happens if the user changes the device time?